2020-02-Spring/4_graphs/README.md

```diff
@@ -119,21 +175,170 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
+### 2. SOLUTION 2
+
+Instead of validating the status of current vertex by checking vertices(stations) it relies on, we could approach the problem top-down: we assume that all stations are shut down. Starting with known power plant thats remain active, we activate adjacent stations and substations of the current active station using DFS.Stations that are not activated by the end of the algorithm are the ones affected by shutting down K stations.
+
+<details>
+<summary>Click to see the algorithm illustration</summary>
+
+![Algorithm in steps](./PowerStations/Alg_illustration.png)
+</details>
+
+###### Pseudocode
+```
+void Solve() {
+  -> mark all stations as Offline
+  -> mark power plants that are in maintenance as Offline
+  // start marking from active power plants i-th
+  for (i : M) {
+    if (PowPlants[i] is active)
+      DFS(PowPlants[i]);
+  }
+}
+void DFS(int Cur) {
+  if (Cur is in maintenance) return; // stop checking
+
+  // Otherwise mark Cur as active and visited
+  Visited[Cur] = true;
+  Active[Cur] = true;
+
+  // check its adjacent
+  for (i : N) {
+    // if i is not visited and has path from Cur->i
+    if (!Visited[i] && Matrix[Cur][i])
+      DFS(i);
+  }
+}
+```
+###### Source code
```

Comment:
Source code indentation is off. If this code were to be copy/pasted and compiled, it likely wouldn't work. See your solution files to see what it should look like.